src/layout_abstract_syntax.rs

```diff
@@ -28,12 +29,27 @@ impl LayoutAbstractTag {
 		Self {
 			namespace,
 			name,
+			layout_attributes: Default::default(),
 			attributes: Vec::new(),
 		}
 	}
 
 	pub fn add_attribute(&mut self, attribute: Attribute) {
-		self.attributes.push(attribute);
+		match &attribute.name[..] {
+			// Layout attributes, stored separately
+			"width" => self.layout_attributes.width = attribute.dimension(),
+			"height" => self.layout_attributes.height = attribute.dimension(),
+			"x-align" => self.layout_attributes.x_align = attribute.percent(),
+			"y-align" => self.layout_attributes.y_align = attribute.percent(),
+			"x-padding" => self.layout_attributes.padding.set_horizontal(attribute.dimension()),
+			"y-padding" => self.layout_attributes.padding.set_vertical(attribute.dimension()),
+			"padding" => self.layout_attributes.padding = attribute.box_dimensions(),
+			"x-spacing" => self.layout_attributes.spacing.set_horizontal(attribute.dimension()),
+			"y-spacing" => self.layout_attributes.spacing.set_vertical(attribute.dimension()),
+			"spacing" => self.layout_attributes.spacing = attribute.box_dimensions(),
```

Comment:
Can we move this match to another function and call that function when there is no `:` prefix? That will be cleaner when it comes time to add `on:`-prefixed event handler attributes (and any other future attribute categories).

Reply:
Are also might want to add the `scroll` attribute for later implementation.

Reply:
Good idea, I've refactored that function.